apps/admin.py

```diff
@@ -47,3 +69,72 @@ def field_func(self, obj):
         return field_func
 
     return wrap
+
+
+class SafeJson(str):
+    def __html__(self):
+        return self
+
+
+def pretty_json_print(field, indent=2, sort_keys=True, line_separator=u'\n'):
+    response = json.dumps(field, sort_keys=sort_keys, indent=indent)
+
+    # Get the Pygments formatter
+    formatter = HtmlFormatter(style='colorful', lineseparator=line_separator)
+
+    # Highlight the data
+    response = highlight(response, JsonLexer(), formatter)
+
+    # Get the stylesheet
+    style = "<style>" + formatter.get_style_defs() + "</style><br>"
+
+    # Safe the output
+    return SafeJson(style + response)
+
+
+def object_detail_admin_link(obj):
+    """
+    Returns the url admin link of the object passed in argument
+    """
+    return format_html(
+        '<a href="{}">{}</a>',
+        admin_change_url(obj),
+        obj.id if obj else ''
+    )
+
+
+class ShipmentAdminDisplayMixin:
```

Comment:
Seeing how every time this is used, you set `list_per_page = settings.ADMIN_PAGE_SIZE` would it make sense to set that setting here in this mixin?

Reply:
We want to keep the concerns separate here - this mixin adds the 'shipment_display' function, for adding shipment links. Doesn't have anything to do with page sizes. I agree it might be nice to DRY this up if it's a setting we want to use globally, but it should be set in either a different mixin, or we can make our own base class derived from ModelAdmin.

Reply:
Good point! But since this is a direct configuration param inherited from `ModelAdmin`, I would rather keep it explicit in any child class.